sns_sqs_communicator/queue/protocol.py

```diff
@@ -0,0 +1,58 @@
+import collections.abc
+import typing
+
+import mypy_boto3_sqs.type_defs
+
+from .. import clients, fifo_attributes_creator
+
+
+class QueueProtocol(typing.Protocol):
+    """Protocol for queue."""
+
+    client: clients.SQSClient
+    queue_url: str
+    fifo_attrs_creator: (
+        fifo_attributes_creator.FifoAttributesCreatorProtocol | None
+    ) = None
```

Comment:
I think this shouldn't be here because it's vendor-related field, it's needed only for sqs queue (or sns topic), so we need to place it to constructor in `SQSQueue` class. Technically other vendors can set up this differently or even don't need to setup this